staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/forbidden/doc.go

```diff
@@ -32,13 +32,11 @@ type T1 struct {
 	// +forbidden
 	PS *string `json:"ps"`
 
-	// +forbidden
 	T2 T2 `json:"t2"`
 	// +forbidden
 	PT2 *T2 `json:"pt2"`
 }
```

Comment:
For followup, we should added test cases for `[]string`, `[]*string`, `[]T2`, `[]*T2`, `map[string]string`, `map[string]T2`, and probably `type AliasOfSlice []string`.